client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -98,22 +101,35 @@ private AckFuture writeEventInternal(String routingKey, Type event) {
                 segmentWriter = selector.getSegmentOutputStreamForKey(routingKey);
             }
             try {
-                segmentWriter.write(new PendingEvent(routingKey, data, result));
+                segmentWriter.write(new PendingEvent(routingKey, data, ackFuture));
             } catch (SegmentSealedException e) {
                 log.info("Segment was sealed: {}", segmentWriter);
                 handleLogSealed(Segment.fromScopedName(segmentWriter.getSegmentName()));
             }
         }
-        return new AckFutureImpl(result, () -> {
-            if (!closed.get()) {
-                flushInternal();
+
+        //The goal is to remove the need of flushInternal here. i.e. FlushInternal is used to handle segment
+        //sealed exception which should now be performed on the internalExecutor.
+        CompletableFuture<Void> result = new CompletableFuture<>();
```

Comment:
The CompletableFuture `ackFuture ` is completed on a netty thread e.g: `epollEventLoopGroup-4-1`
So, if we return this future it is possible that the client can can chain on this future and this work can be scheduled on this thread. Hence the pattern of creating a new Completablefuture has been used.